src/Compiler.php

```diff
@@ -2472,16 +2477,18 @@ protected function compileImport($rawPath, OutputBlock $out, $once = false)
         if ($rawPath[0] === Type::T_STRING) {
             $path = $this->compileStringContent($rawPath);
 
-            if (strpos($path, 'url(') !== 0 && $path = $this->findImport($path)) {
-                if (! $once || ! \in_array($path, $this->importedFiles)) {
-                    $this->importFile($path, $out);
-                    $this->importedFiles[] = $path;
+            if (strpos($path, 'url(') !== 0 && $filePath = $this->findImport($path, $this->currentDirectory)) {
+                if (! $once || ! \in_array($filePath, $this->compilationResult->getImportedFiles())) {
+                    $this->importFile($filePath, $out);
+                    $this->compilationResult->addImportedFile($this->currentDirectory, $path, $filePath);
                 }
 
                 return true;
             }
 
-            $this->appendRootDirective('@import ' . $this->compileImportPath($rawPath) . ';', $out);
+            $path = $this->compileImportPath($rawPath);
+            $this->appendRootDirective('@import ' . $path . ';', $out);
+            $this->compilationResult->addIncludedFile($path);
```

Comment:
this looks wrong to me. This file is not included as it is compiled to a CSS `@import` (and might not even be a file that exists on disk)

Reply:
I will dig again, but as long I can remember I sticked to what dart-saas was doing on this files liste, even if it doesn't feels what you are expecting.

Reply:
dart-sass is only adding values in this file list for sass imports, not for CSS imports (which are compiled by `visitStaticImport`)

Reply:
the doc about filesystem import vs URI is related to the fact that dart-sass supports using URIs for imports, not only paths (when running the dart API, it can support `package:` URIs which resolve to a dart package installed in the project for instance)